File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        int hi = nums.length - 1;
        int lo = 0;

        while( lo < hi ){
            if (nums[lo] < nums[hi]) return nums[lo];
            int mid = lo + (hi - lo)/2;
            if ( nums[mid] > nums[hi]) lo = mid + 1;
            else hi = mid;
        }
        return nums[lo];
    }

}

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.js
class Solution {
    maxSlidingWindow(nums, k) {
        // moronic approach is using DeQueue
        let res = [];
        let deq = [];
        
        for ( let i = 0; i < nums.length; i++ ){
            if ( deq.length && deq[0] <= i - k ) deq.shift();
            while ( deq.length && nums[deq[deq.length - 1]] < nums[i]) deq.pop();
            deq.push(i);
            if(i >= k - 1) res.push(nums[deq[0]]);
        } return res;
    }
}
